pkg/gateway/operations/putbucket.go

```diff
@@ -0,0 +1,32 @@
+package operations
+
+import (
+	"net/http"
+
+	gatewayerrors "github.com/treeverse/lakefs/pkg/gateway/errors"
+	"github.com/treeverse/lakefs/pkg/permissions"
+)
+
+// PutBucket handles S3 Create Bucket operations.  It does *not* actually
+// create new repos (there is not enough information in the S3 request to
+// create a new repo), but *does* detect whether the repo already exists.
+type PutBucket struct{}
+
+func (controller *PutBucket) RequiredPermissions(_ *http.Request, repoID string) ([]permissions.Permission, error) {
+	return []permissions.Permission{
+		{
+			Action:   permissions.ReadRepositoryAction,
```

Comment:
use `CreateRepositoryAction`

Reply:
Yikes!  You're right, S3 only accepts a single permission.  So if I don't have `CreateBucket` permission on S3 and I try to `create-bucket`, I get 403 (AccessDenied) rather than 409 (BucketAlreadyOwnedByYou).

Switching to match. :shrug: